### PR TITLE
Fix broken links in README

### PR DESCRIPTION
**Pull request by GenevieveBuckley**
# Description

This PR fixes several broken links in the README, and removes a reference to an out of date project roadmap (roadmap for 0.3 releases).

## Type of change

- [x] This change is a documentation update

# References

None

# How has this been tested?

- [x] checked new URLs do not produce a 404 error

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [ ] ~~I have commented my code, particularly in hard-to-understand areas~~
- [x] I have made corresponding changes to the documentation
- [ ] ~~I have added tests that prove my fix is effective or that my feature works~~

### CODE REVIEW

File: README.md
Comment (jni) [inline README.md:139]:
```suggestion
For more information about our plans for `napari` you can read our [mission and values statement](https://napari.org/docs/dev/developers/mission_and_values.html), which includes more details on our vision for supporting a plugin ecosystem around napari. We also have a [roadmap](https://napari.org/docs/dev/developers/roadmap_0_4.html) that captures the current development priorities within the project.
```

**Review by jni — APPROVED** [review]
Thanks @GenevieveBuckley! I've made a suggestion pointing to the current roadmap, but otherwise this looks good to me!

File: README.md
Comment (GenevieveBuckley) [reply README.md:139]: Nope, I don't wanna do that! It'll just be out of date soon too.

What I would be ok with is for this file to be named something like "current_roadmap", and then when we make a release someone can copy paste the old contents into "roadmap_0_whatever" to archive.

Comment (jni) [reply README.md:139]: We probably need a system to deal with out of date links, but I'd say removing the roadmap from the readme is not the solution. We definitely want to point people to the roadmap!

Comment (sofroniewn) [reply README.md:139]: I like the idea of a `roadmap_latest` or `roadmap_current` alias. Maybe we accept the suggestion from @jni for now as I agree keeping to roadmap link on the homepage is good, and we then ask @kne42 to setup an alias system somewhere or look into some other solution?

Comment (GenevieveBuckley) [reply README.md:139]: `roadmap_latest` is fine by me. 

Why bother with an alias? Why not just use `roadmap_latest` for whatever is being worked on now, and then archive the contents after a major release when you want to start a new roadmap? This seems less prone to breaking, at least to me.

Comment (sofroniewn) [reply README.md:139]: I was thinking you'd want the roadmap to be present at both urls, but maybe that's not needed. It also seemed much easier to me to change the alias when the new roadmap is ready rather than having to change the actual location of the old roadmap, but I'm not sure. I'd defer to @kne42 here

Comment (GenevieveBuckley) [reply README.md:139]: I'm going to open a new issue for the roadmap link, so that it doesn't hold up this PR: https://github.com/napari/napari/issues/2277

**Pull request merged**
